### DISCUSSION TITLE
setting up the architecture

### DISCUSSION THREAD

**Discussion by ajnaf** [ajnaf/flutter_course#3] [General]
folders اول خطوة نقوم بإنشاء ال 
1 - app
2 - models
3 - services
4 - ui 
        a- shared
        b- views
        c- widgets

==========================================
ثم نقوم باضافة المكاتب اللازمة داخل pubspec.yaml
في الdependencies 
نقوم باضافة التالي
  
  get_it: ^5.0.6
  stacked: ^1.9.5
  stacked_services: ^0.7.2
  injectable: ^1.0.7

اما داخل ال dev_dependencies

  build_runner: ^1.10.11
  injectable_generator: ^1.0.7
  stacked_generator: 

=====================================

**Comment by ajnaf** [comment]
ثانيا :
داخل ال app folder
قم بتكوين
app.dart
locator.dart
=======================
app.dart بالنسبة الى 
الخاصة بالابليكيشون views سيتم فيه اضافة ال
=======================
locator.dart بالنسبة الى 
سيبقى كما هو دائما
=======================

**Reply by nor-nor** [reply]
السلام عليكم
باش مهندس احمد 
انت الملفات 

1 - app
2 - models
3 - services
4 - ui
a- shared
b- views
c- widgets

كونتهم داخل ملف  الـ lib
هل اجباري ولا كان انا كونتهم فى ROOTالرئيسي عادي ؟

**Reply by nor-nor** [reply]
انا درت زي مانت مداير كونتهم داخل ملف ال
lib

**Reply by nor-nor** [reply]
![image](https://user-images.githubusercontent.com/80286874/110880018-9fcc7e00-82e6-11eb-8183-563726ece422.png)

**Reply by nor-nor** [reply]
انا حاليا نخدم علي ملف القديم وضفتله كل الي قلت عليه 
والحمد لله يشتغل مافيش مشاكل

**Reply by ajnaf** [reply]
نعم يتم اضافتهم داخل الlib 
الملف الوحيد خارج الlib هو الassets

**Comment by ajnaf** [comment]
ثالثا نقوم باضافة 
  configureInjection();
main() داخل دالة ال
======================================

**Comment by ajnaf** [comment]
رابعا نقوم باضافة بعض الخصائص داخل 
materialApp 
ليصبح
==========================
MaterialApp(
      title: 'الشركة العامة للكهرباء',
      initialRoute: Routes.startupView,
      onGenerateRoute: StackedRouter().onGenerateRoute,
      navigatorKey: StackedService.navigatorKey,
      theme: ThemeData(
        primarySwatch: Colors.green,
      ),
    );

=============================
نلاحظ ان هناك خطا 
اولا نقوم بعمل import 
للملفات المفقودة 
وذلك بالتأشير على الخطا ثم الضغط على  ctrl+.
سيطلب منك اختيار الملف لعمل import
![image](https://user-images.githubusercontent.com/31656833/110804333-c0fe8180-8288-11eb-9127-538fed3039eb.png)

**Reply by ajnaf** [reply]
في حال عدم ظهور الimport 
يعني هذا الملف لم يتكون بعد

**Comment by ajnaf** [comment]
نقوم باضافة الview , viewmodel 
ثم نقوم باعطاء الامر التالي في
الTerminal
flutter packages pub run build_runner build
ليقوم بانشاء الملفات المطلوبة
ملاحظة
يتم كتابة هذا الامر كل ما يتم اضافة view ,viewmodel
بالاضافة لا ننسى اضافة الview 
داخل ال app.dart
